odbc/src/api/functions.rs

```diff
@@ -832,7 +831,7 @@ pub unsafe extern "C" fn SQLDisconnect(connection_handle: HDbc) -> SqlReturn {
 
 fn sql_driver_connect(conn: &Connection, odbc_uri_string: &str) -> Result<MongoConnection> {
     let mut odbc_uri = ODBCUri::new(odbc_uri_string)?;
-    let mongo_uri = odbc_uri.remove_to_mongo_uri()?;
+    let client_options = odbc_uri.try_into_client_options()?;
     let auth_src = odbc_uri.remove_or_else(|| "admin", &["auth_src"]);
```

Comment:
To make sure that the KEYWORD list and the options to retrieve are the same, could we have the options/keywords specified as const str and then the regex for parsing the ODBC connection string would use them, and also the odbc_uri remove calls. This was we won't end up with discrepancies between the 2.